packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/explore/DataExplorer.tsx

```diff
@@ -1,39 +1,40 @@
 "use client";
 
-import { Loader } from "lucide-react";
 import { useSearchParams } from "next/navigation";
-import { useQuery } from "@tanstack/react-query";
+import { anvil } from "viem/chains";
+import { useEffect, useState } from "react";
+import { useChain } from "../../../../../../hooks/useChain";
+import { useTableId } from "../../../../../../hooks/useTableId";
+import { useDeployedTablesQuery } from "../queries/useDeployedTablesQuery";
+import { useTableDataQuery } from "../queries/useTableDataQuery";
+import { SQLEditor } from "./SQLEditor";
 import { TableSelector } from "./TableSelector";
 import { TablesViewer } from "./TablesViewer";
 
 export function DataExplorer() {
+  const { id: chainId } = useChain();
+  const [query, setQuery] = useState("");
   const searchParams = useSearchParams();
-  const { data: tables, isLoading } = useQuery({
-    queryKey: ["tables"],
-    queryFn: async () => {
-      const response = await fetch("/api/tables");
-      const json = await response.json();
-      if (!response.ok) {
-        throw new Error(json.error);
-      }
+  const { data: deployedTables } = useDeployedTablesQuery();
+  const selectedTableId = searchParams.get("tableId") ?? deployedTables?.[0]?.tableId;
+  const deployedTable = deployedTables?.find(({ tableId }) => tableId === selectedTableId);
+  const tableId = useTableId(deployedTable);
+  const { data: tableData, isLoading } = useTableDataQuery({ deployedTable, query });
 
-      return json;
-    },
-    select: (data) => data.tables.map((table: { name: string }) => table.name),
-    refetchInterval: 15000,
-    throwOnError: true,
-    retry: false,
-  });
-  const selectedTable = searchParams.get("tableId") || (tables?.length > 0 ? tables[0] : null);
-
-  if (isLoading) {
-    return <Loader className="animate-spin" />;
-  }
+  useEffect(() => {
+    if (!deployedTable || !tableId) return;
+    if (chainId === anvil.id) {
+      setQuery(`SELECT * FROM "${tableId}"`);
```

Comment:
The handling of queries is still in question. At the moment it's being passed as a parameter but will adapt once there is a clearer consensus on how to handle it.